orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java

```diff
@@ -107,4 +110,47 @@ public LLMChoice getChoice() {
         .getChoices()
         .get(0);
   }
+
+  /**
+   * Transform a JSON response into an entity of the given type.
+   *
+   * <p>This is possible on a request with a {@link OrchestrationTemplate#withJsonSchemaResponse}
+   * configured into {@link OrchestrationModuleConfig#withTemplateConfig}.
+   *
+   * @param type the class type to deserialize the JSON content into.
+   * @return the deserialized entity of type T.
+   * @param <T> the type of the entity to deserialize to.
+   * @throws OrchestrationClientException if the model refused to answer the question or if the
+   *     content
+   */
+  @Nonnull
+  public <T> T asEntity(@Nonnull final Class<T> type) throws OrchestrationClientException {
+    final String refusal =
+        ((LLMModuleResultSynchronous) getOriginalResponse().getOrchestrationResult())
+            .getChoices()
+            .get(0)
+            .getMessage()
+            .getRefusal();
+    if (refusal != null) {
+      throw new OrchestrationClientException(
+          "The model refused to answer the question: " + refusal);
+    }
+    try {
+      return new ObjectMapper().readValue(getContent(), type);
+    } catch (InvalidDefinitionException e) {
+      throw new OrchestrationClientException(
+          "Failed to deserialize the JSON content. Please make sure to use the correct class and that the class has a no-args constructor and it is static: "
+              + e.getMessage()
+              + "\nJSON content: "
```

Comment:
(Minor)

New-line in log messages may be a problem. Usually considered bad practice.
But You are dumping the whole content into the exception message without sanitization anyway.

Reply:
This will be [discussed in slack](https://sap-ml.slack.com/archives/C070WHEH35L/p1749113950997649).